src/h5web/dataset-visualizer/VisProvider.tsx

```diff
@@ -32,19 +31,29 @@ function VisProvider(props: Props): ReactElement {
   const rawValues = useValue(dataset.id);
   const rawDims = isSimpleShape(dataset.shape) ? dataset.shape.dims : [];
 
-  const xDim = mapperState ? mapperState.indexOf('x') : -1;
-
   const mappedValues = useMemo(() => {
     if (!rawValues) {
       return undefined;
     }
 
-    if (rawDims.length === 2 && xDim === 0) {
-      return transpose(rawValues);
+    if (rawDims.length !== 0) {
+      return ndarray(rawValues.flat(), rawDims);
     }
 
     return rawValues;
-  }, [rawValues, rawDims.length, xDim]);
+  }, [rawValues, rawDims]);
+
+  const values = useMemo(() => {
+    if (mapperState === undefined) {
+      return mappedValues;
+    }
+    const slicingIndices = mapperState.map(val => (isNumber(val) ? val : null));
```

Comment:
`slicingIndices` are `null` for 'x' and 'y' so that the full dimension is taken (see https://github.com/scijs/ndarray#arraypickp0-p1-)